File: 프로그래머스/lv1_완주하지_못한_선수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";

    multiset<string> mset;

    for ( auto iter = participant.begin(); iter != participant.end(); iter++ ){
        mset.insert(*iter);
    }

    for ( auto iter = completion.begin(); iter != completion.end(); iter++){
        mset.erase(mset.find(*iter));
    }

    answer = *(mset.begin());

    return answer;
}

int main(void){

    string ret;
    vector<string> participant = {"leo", "kiki", "eden", "leo"};
    vector<string> completion = {"eden", "kiki", "leo"};
    
    ret = solution(participant, completion);

    printf("%s\n", ret.c_str());

    return 0;
}

File: 프로그래머스/lv1_완주하지_못한_선수.py

def solution(participant, completion):
    answer = ''

    hashtable = {}
    for people in participant:
        if people in hashtable:
            hashtable[people] += 1
        else:
            hashtable[people] = 1

    for people in completion:
        hashtable[people] -= 1

    for people in hashtable:
        if hashtable[people] != 0:
            answer = people
    
    print(hashtable)

    return answer

if __name__ == "__main__":

    participant = ["leo", "kiki", "eden"]
    completion = ["eden", "kiki"]
    ret = solution( participant, completion )
    print ( ret )